src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1317,8 +1317,13 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
     void visit_Return(const AST::Return_t &x) {
         std::string return_var_name = "_lpython_return_variable";
         if(current_scope->scope.find(return_var_name) == current_scope->scope.end()) {
-            throw SemanticError("Return type of function is not defined",
+            if (x.m_value) {
+                throw SemanticError("Return type of function is not defined",
                                 x.base.base.loc);
+            }
+            // this may be a case with void return type (like subroutine)
```

Comment:
We should add a check here (we can do that in a subsequent PR) that we are actually in a subroutine, otherwise give an error message.

Reply:
Yes!